gadget/gadget.go

```diff
@@ -213,6 +221,82 @@ func (vs *VolumeStructure) IsPartition() bool {
 	return vs.Type != "bare" && vs.Role != schemaMBR
 }
 
+// IsFixedSize tells us if size is fixed or if there is range.
+func (vs *VolumeStructure) IsFixedSize() bool {
+	return vs.Size == vs.MinSize
+}
+
+func MinStructureOffset(vss []VolumeStructure, idx int) quantity.Offset {
+	if vss[idx].Offset != nil {
+		return *vss[idx].Offset
+	}
+	// Move up in the slice for minimum: the minimum offset will be
+	// the first fixed offset that we find while going up plus all
```

Comment:
don't understand the up here,  we are going backward down the structure actually